sklearn/compose/_column_transformer.py

```diff
@@ -235,7 +246,7 @@ def _iter(self, X=None, fitted=False, replace_strings=False):
                 elif trans == 'drop':
                     continue
 
-            yield (name, trans, sub, get_weight(name))
+            yield (name, trans, sub, get_weight(name), column)
```

Comment:
to use `_iter` in `_update_fitted_transformers` I needed to add `column` in the output (as it is needed there), so therefore you see also changes in the diff for each place where `_iter` is called now, making the diff a bit bigger.

Reply:
FWIW (cosmetics), I don't think we any longer need to have X passed into this method, or its subset passed 
out. That feature is only used in one method (`_fit_transform`) where we could instead just call `_get_columns`.

Reply:
Good idea! Made the small change